### PR TITLE
chore(*) update for newer versions of kong

### PR DESCRIPTION
**Pull request by gszr**
## Summary

Update for newer versions of Kong (>= 0.15).

- Update plugin schema to new DAO schema format
- Update tests - use blueprints, remove usage of APIs entity
- Bonus: add .luacheckrc

The changes were tested with a local OpenWhisk environment, following
instructions contained in `specs/access_spec.lua`.

### CODE REVIEW

**Comment by devadvocado** [comment]
hey @bungle, was hoping to get your review on this real quick since someone was IBM was hoping to see the plugin in action. Thanks!

**Comment by rabbah** [comment]
Are there any plans to update for openwhisk webactions which allow for more general serverless APIs? https://github.com/Kong/kong-plugin-openwhisk/issues/5

File: kong/plugins/openwhisk/schema.lua [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
-    port          = { type = "number",  default  = 443   },
-    path          = { type = "string",  required = true  },
-    action        = { type = "string",  required = true  },
-    https         = { type = "boolean", default  = true  },
-    https_verify  = { type = "boolean", default  = false },
-    result        = { type = "boolean", default  = true  },
+    config = {
+      type = "record",
+      fields = {
+        { timeout       = { type = "integer",  default  = 60000 } },
```
Comment (bungle) [inline kong/plugins/openwhisk/schema.lua]:
```suggestion
        { timeout       = { type = "integer", default  = 60000 } },
```

File: kong/plugins/openwhisk/schema.lua [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
+        { keepalive     = { type = "integer",  default  = 60000 } },
```
Comment (bungle) [inline kong/plugins/openwhisk/schema.lua]:
```suggestion
        { keepalive     = { type = "integer", default  = 60000 } },
```

File: kong/plugins/openwhisk/schema.lua [hunk 3]
```diff
@@ -1,14 +1,20 @@
...
+        { service_token = { type = "string"                     } },
+        { host          = { type = "string",  required = true   } },
+        { port          = { type = "integer",  default  = 443   } },
```
Comment (bungle) [inline kong/plugins/openwhisk/schema.lua]:
```suggestion
        { port          = { type = "integer", default  = 443   } },
```

**Comment by hbagdi** [comment]
ping @gszr

**Pull request merged**